src/App.tsx

```diff
@@ -1,26 +1,104 @@
 import React from 'react';
+import { useEffect, useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoCreate } from './components/TodoCreate/TodoCreate';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoServices from './api/todos';
+import { Todo } from './types/Todo';
+import { StatusFilter } from './types/StatusFilter';
+import { filterTodos } from './utils/filterTodos';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [statusFilter, setStatusFilter] = useState<StatusFilter>(
+    StatusFilter.All,
+  );
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.NO_ERROR);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const sendError = (error: ErrorType) => {
+    setErrorType(error);
+
+    const timer = setTimeout(() => {
+      setErrorType(ErrorType.NO_ERROR);
+
+      clearTimeout(timer);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const todos = await todoServices.getTodos();
+
+        setTodoList(todos);
+      } catch {
+        sendError(ErrorType.LOAD_TODOS);
+      }
+    })();
+  }, []);
+
+  const filteredTodos = filterTodos(statusFilter, todoList);
+  const activeTodosCount = todoList.filter(todo => !todo.completed).length;
+  const hasCompletedTodos = todoList.length > activeTodosCount;
+
+  const clearCompleted = async () => {
+    const completedTodos = todoList.filter(todo => todo.completed);
+
+    for (const todo of completedTodos) {
+      try {
+        await todoServices.deleteTodo(todo.id);
+        setTodoList(currentTodoList =>
+          currentTodoList.filter(currentTodo => currentTodo.id !== todo.id),
+        );
+      } catch (error) {
+        setErrorType(ErrorType.DELETE_TODO);
+      }
+    }
+  };
+
+  if (!todoServices.USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoCreate
+          setTodoList={setTodoList}
+          sendError={sendError}
+          userId={todoServices.USER_ID}
+          setTempTodo={setTempTodo}
+          todoList={todoList}
+        />
+
+        {todoList.length > 0 && (
```

Comment:
will be more readable i think

```suggestion
        {!!todoList.length && (
```